## Импортируем необходимые библиотеки.

In [1]:
import pandas as pd
import numpy as np

## Прочитаем датасет в переменную.

In [2]:
df = pd.read_csv('cian_dataset.csv')
pd.options.display.max_columns = None

## Проанализируем датасет.

In [3]:
df.head(3)

Unnamed: 0  ID  объявления Количество комнат       Тип  \
0           0       271271157                 4  Квартира   
1           1       271634126                 4  Квартира   
2           2       271173086   4, Оба варианта  Квартира   

                          Метро                          Адрес  \
0  м. Смоленская (9 мин пешком)  Москва, улица Новый Арбат, 27   
1  м. Смоленская (8 мин пешком)  Москва, улица Новый Арбат, 27   
2  м. Смоленская (7 мин пешком)  Москва, улица Новый Арбат, 27   

       Площадь, м2                        Дом   Парковка  \
0       200.0/20.0           5/16, Монолитный  подземная   
1  198.0/95.0/18.0  5/16, Монолитно-кирпичный  подземная   
2  200.0/116.0/4.0                       5/16  подземная   

                                                Цена  \
0  500000.0 руб./ За месяц, Залог - 500000 руб., ...   
1  500000.0 руб./ За месяц, Залог - 500000 руб., ...   
2  500000.0 руб./ За месяц, Залог - 500000 руб., ...   

                     Телефоны  \
0                +79166369231   
1                +79850608590   
2  +79672086536, +79099269384   

                                            Описание        Ремонт  \
0  Без комиссии для нанимателя! Бонус коллегам 12...  Дизайнерский   
1  Лот 93107. Елена Анисимова.\n\nБонус агенту 50...  Дизайнерский   
2  ID 36380:  Шикарная 4-х км. квартира в ЖК "Нов...    Евроремонт   

  Площадь комнат, м2 Балкон             Окна                          Санузел  \
0                NaN    NaN              NaN                              NaN   
1        25 25 20 25    NaN  На улицу и двор  Совмещенный (2), Раздельный (1)   
2                NaN    NaN  На улицу и двор                  Совмещенный (3)   

            Можно с детьми/животными  \
0  Можно с детьми, Можно с животными   
1                     Можно с детьми   
2                     Можно с детьми   

                                       Дополнительно        Название ЖК  \
0  Мебель в комнатах, Мебель на кухне, Ванна, Душ...  Новый Арбат, 2010   
1  Мебель в комнатах, Мебель на кухне, Ванна, Душ...        Новый Арбат   
2  Мебель в комнатах, Мебель на кухне, Ванна, Душ...        Новый Арбат   

  Серия дома  Высота потолков, м                Лифт Мусоропровод  \
0        NaN                 3.0  Пасс (4), Груз (1)           Да   
1        NaN                 3.5  Пасс (1), Груз (1)          Нет   
2        NaN                 3.2            Пасс (1)          NaN   

                      Ссылка на объявление  
0  https://www.cian.ru/rent/flat/271271157  
1  https://www.cian.ru/rent/flat/271634126  
2  https://www.cian.ru/rent/flat/271173086

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23368 entries, 0 to 23367
Data columns (total 25 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                23368 non-null  int64  
 1   ID  объявления            23368 non-null  int64  
 2   Количество комнат         22327 non-null  object 
 3   Тип                       23368 non-null  object 
 4   Метро                     22053 non-null  object 
 5   Адрес                     23368 non-null  object 
 6   Площадь, м2               23368 non-null  object 
 7   Дом                       23368 non-null  object 
 8   Парковка                  9951 non-null   object 
 9   Цена                      23368 non-null  object 
 10  Телефоны                  23368 non-null  object 
 11  Описание                  23368 non-null  object 
 12  Ремонт                    20613 non-null  object 
 13  Площадь комнат, м2        14458 non-null  object 
 14  Балкон

In [5]:
df.isna().sum()

Unnamed: 0                      0
ID  объявления                  0
Количество комнат            1041
Тип                             0
Метро                        1315
Адрес                           0
Площадь, м2                     0
Дом                             0
Парковка                    13417
Цена                            0
Телефоны                        0
Описание                        0
Ремонт                       2755
Площадь комнат, м2           8910
Балкон                       7978
Окна                         6613
Санузел                      2672
Можно с детьми/животными     6096
Дополнительно                 357
Название ЖК                 17520
Серия дома                  21205
Высота потолков, м          12162
Лифт                         5500
Мусоропровод                10522
Ссылка на объявление            0
dtype: int64

In [6]:
df.describe()

Unnamed: 0  ID  объявления  Высота потолков, м
count  23368.000000    2.336800e+04         11206.00000
mean   11683.500000    2.676265e+08             2.98345
std     6745.904881    1.876862e+07             7.61472
min        0.000000    1.072986e+08             1.20000
25%     5841.750000    2.712663e+08             2.64000
50%    11683.500000    2.739030e+08             2.64000
75%    17525.250000    2.746872e+08             2.80000
max    23367.000000    2.750064e+08           320.00000

Нужно:
- Unnamed: 0 - удалить
- ID  объявления - удалить
- Количество комнат - избавиться от слов и перевести в тип int (например "4, Оба варианта"), 1041 пропущенных значений
- Тип - удалить, так как одно единственное значение - квартира;
- Метро - разбить на 2 столбца, разделить метро и расстояние до него, 1315 пропущенных значений;
- Адрес - пока выглядит ок
- Площадь, м2 - разбить на три столбца
- Дом - разбить на три столбца (текущий этаж, общая этажность, тип дома);
- Парковка - 13417 пропущенных значений;
- Цена - подумать как разбить
- Телефоны - удалить
- Описание - удалить
- Ремонт - 2755 пропущенных значений;
- Площадь комнат, м2 - удалить;
- Балкон - разбить на балкон и лоджию (пример: "Балкон (1), Лоджия (1)"), 7978 пропущенных значений;
- Окна - разбить на улицу и двор, 6613 пропущенных значений;
- Санузел - разбить на совмещенный и раздельный, 2672 пропущенных значений;
- Можно с детьми/животными - разбить на 3, 6096 пропущенных значений;
- Дополнительно - слишком много всего, в первом приближении можно удалить;
- Название ЖК - удалить, 17520 пропущенных значений;
- Серия дома - думаю можно удалить, 21205 пропущенных значений;
- Высота потолков, м - 12162 пропущенных значений, слишком много. Удалим столбец;
- Лифт - удалить;
- Мусоропровод - 10522 пропущенных значений;
- Ссылка на объявление - удалить.       

## Работа над датасетом.

### 1. Удалим ненужные столбцы:

In [7]:
df.drop(columns=['Unnamed: 0', 'ID  объявления', 'Тип', 'Телефоны', 'Название ЖК', 'Серия дома', 'Ссылка на объявление', 'Описание', 'Дополнительно', 'Площадь комнат, м2', 'Лифт', 'Высота потолков, м'], axis=1, inplace=True)

In [8]:
df.head(3)

Количество комнат                         Метро  \
0                 4  м. Смоленская (9 мин пешком)   
1                 4  м. Смоленская (8 мин пешком)   
2   4, Оба варианта  м. Смоленская (7 мин пешком)   

                           Адрес      Площадь, м2                        Дом  \
0  Москва, улица Новый Арбат, 27       200.0/20.0           5/16, Монолитный   
1  Москва, улица Новый Арбат, 27  198.0/95.0/18.0  5/16, Монолитно-кирпичный   
2  Москва, улица Новый Арбат, 27  200.0/116.0/4.0                       5/16   

    Парковка                                               Цена        Ремонт  \
0  подземная  500000.0 руб./ За месяц, Залог - 500000 руб., ...  Дизайнерский   
1  подземная  500000.0 руб./ За месяц, Залог - 500000 руб., ...  Дизайнерский   
2  подземная  500000.0 руб./ За месяц, Залог - 500000 руб., ...    Евроремонт   

  Балкон             Окна                          Санузел  \
0    NaN              NaN                              NaN   
1    NaN  На улицу и двор  Совмещенный (2), Раздельный (1)   
2    NaN  На улицу и двор                  Совмещенный (3)   

            Можно с детьми/животными Мусоропровод  
0  Можно с детьми, Можно с животными           Да  
1                     Можно с детьми          Нет  
2                     Можно с детьми          NaN

### 2. Количество комнат: 

#### 2.1. Удалим лишние слова, и оставим только количество

In [9]:
#df['Количество комнат'].value_counts()

In [10]:
df['Количество комнат'] = df['Количество комнат'].astype('str')
df['Количество комнат'].dtype

dtype('O')

In [11]:
df[['Количество комнат', 'Тип комнат']] = df['Количество комнат'].str.split(',', expand=True)
df.drop(columns='Тип комнат', inplace=True)

#### 2.2. Обработаем пропуски:

In [12]:
df['Количество комнат'].value_counts().sort_values()

6       138
5       367
nan    1041
4      1163
3      4262
1      7931
2      8466
Name: Количество комнат, dtype: int64

Посмотрим, какой площади соответствуют данные квартиры:

In [13]:
df[df['Количество комнат'] == 'nan']['Площадь, м2']

115                41.0
120                63.0
270            26.0/5.0
288                27.0
289       27.7/13.4/6.8
              ...      
23334              23.0
23348              35.0
23353    45.0/22.0/15.0
23362     25.0/17.0/4.0
23363     35.0/16.4/8.0
Name: Площадь, м2, Length: 1041, dtype: object

Предположим, что nan имеет отношение к студиям и заменим пропущенные значения на 0.

In [14]:
df['Количество комнат'] = df['Количество комнат'].replace('nan', '0')
#df['Количество комнат'].value_counts().sort_values()

### 3. Метро:

Разделим на два столбца 'Метро' и 'Расстояние до метро, мин пешком':

In [15]:
df[['Метро', 'Расстояние до метро, мин пешком']] = df['Метро'].str.split('(', expand=True)
df['Метро'] = df['Метро'].str.strip('м. ')
df['Расстояние до метро, мин пешком'] = df['Расстояние до метро, мин пешком'].str[0]

In [16]:
df['Расстояние до метро, мин пешком'].value_counts()

1    8960
5    2140
2    1884
7    1514
3    1409
4    1365
8    1283
6    1244
9    1115
N    1040
0      99
Name: Расстояние до метро, мин пешком, dtype: int64

In [17]:
df['Расстояние до метро, мин пешком'].isna().sum()

1315

В столбце 'Расстояние до метро, мин пешком' заменим пропущенные значения путем заполнения медианными значениями соответствущего метро.

In [18]:
df['Расстояние до метро, мин пешком'] = df['Расстояние до метро, мин пешком'].replace('N', np.nan)

In [19]:
df['Расстояние до метро, мин пешком'].isna().sum()

2355

In [20]:
for metro in df['Метро'].unique():
    median = df.loc[(df['Метро'] == metro), 'Расстояние до метро, мин пешком'].median()
    df.loc[(df['Расстояние до метро, мин пешком'].isna()) & (df['Метро'] == metro), 'Расстояние до метро, мин пешком'] = median

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/numpy/lib/nanfunctions.py:1215: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/numpy/lib/nanfunctions.py:1215: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)


In [21]:
df['Расстояние до метро, мин пешком'].isna().sum()

1317

Для 1317 объявлений не смогли рассчитать расстояние по медианному значению соответствующего метро.

Оставшиеся пропуски заполним просто медианой по всем объявлениям:

In [22]:
df[df['Расстояние до метро, мин пешком'].isna()] = df['Расстояние до метро, мин пешком'].median()
#df['Расстояние до метро, мин пешком'].isna().sum()

Приведем все значения к типу float:

In [23]:
df['Расстояние до метро, мин пешком'] = df['Расстояние до метро, мин пешком'].apply(lambda x: float(x))

### 4. Площадь

In [24]:
df['Площадь, м2'].value_counts()

2.0                1317
40.0/20.0/10.0      223
45.0/30.0/6.0       107
40.0                 99
40.0/20.0/9.0        77
                   ... 
302.9                 1
127.0/69.4/14.0       1
95.0/73.0/12.0        1
97.0/51.0/12.0        1
90.0/48.2/15.5        1
Name: Площадь, м2, Length: 10860, dtype: int64

Предположим, что значения предствляют из себя 'Общая площадь'/'Жилая площадь'/'Площадь кухни', разделим на соответствующие столбцы:

In [25]:
df[['Общая площадь', 'Жилая площадь', 'Площадь кухни']] = df['Площадь, м2'].str.split('/', expand=True)

Из-за очевидной высокой корреляции и наличии большого количества пропусков дропнем столбцы 'Жилая площадь' и 'Площадь кухни', и изначальную 'Площадь, м2'.

In [26]:
df.drop(columns=['Жилая площадь', 'Площадь кухни', 'Площадь, м2'], inplace=True)

Приведем значения к типу float:

In [27]:
df['Общая площадь'] = df['Общая площадь'].apply(lambda x: float(x))

### 5. Дом

Разделим столбец на 'Этаж' и 'Тип дома'.

In [28]:
df[['Этаж', 'Тип дома']] = df['Дом'].str.split(',', expand=True)
df.drop(columns='Дом', inplace=True)

Проверим объявления без типа дома:

In [29]:
df[df['Тип дома'].isna()].shape

(4860, 15)

Заполним пропуски по адресу дома:

### 6. Парковка

In [30]:
df['Парковка'].value_counts()

наземная          5746
подземная         2735
2.0               1317
открытая          1002
многоуровневая     110
на крыше             1
Name: Парковка, dtype: int64

Если парковка есть - 1, пропущенные значения - 0.

In [31]:
df['Парковка'].isnull().sum()

12457

In [32]:
df.loc[df['Парковка'].isin(['наземная', 'подземная', 2.0, 'открытая', 'многоуровневая', 'на крыше']), 'Парковка'] = 1
df.loc[df['Парковка'].isnull(), 'Парковка'] = 0
df['Парковка'].value_counts()

0    12457
1    10911
Name: Парковка, dtype: int64

### 7. Мусоропровод

In [33]:
df['Мусоропровод'].value_counts()

Да     10540
Нет     1683
2.0     1317
Name: Мусоропровод, dtype: int64

Заменим значения 'Да' и '2.0' на 1 (True), и 'Нет' на 0 (False). Отсутствующие значения на 0.

In [34]:
df['Мусоропровод'] = df['Мусоропровод'].replace(['Да', 'Нет', 2.0], [1, 0, 1])
df['Мусоропровод'] = df['Мусоропровод'].fillna(0)

In [35]:
df['Мусоропровод'] = df['Мусоропровод'].apply(lambda x: int(x))
#df['Мусоропровод'].value_counts()

### 8. Цена

In [36]:
 #df['Цена'].value_counts()

Разделим столбец на 'Стоимость аренды', 'Прочее', последний отбросим.

In [37]:
df[['Стоимость аренды', 'Прочее']] = df['Цена'].str.split('/', expand=True)
df.drop(columns=['Прочее', 'Цена'], inplace=True)

Очистим от ' руб.'

In [38]:
df['Стоимость аренды'] = df['Стоимость аренды'].str.strip(' руб.')

In [39]:
df['Стоимость аренды'].value_counts().sort_values()

115700.0        1
10638.0 $       1
1030000.0       1
1475000.0       1
475000.0        1
             ... 
55000.0       950
50000.0      1235
35000.0      1305
45000.0      1441
40000.0      1483
Name: Стоимость аренды, Length: 590, dtype: int64

Видим что остались значения в долларах и евро. Конвертируем их в рубли по курсу 65руб за $ и €.

In [40]:
df['Стоимость аренды'] = df['Стоимость аренды'].astype('str')

for x in df['Стоимость аренды']:
    if x[-1] in ['$', '€']:
        df.loc[x, 'Стоимость аренды'] = float(''.join(x[:-4]))*65

In [41]:
df['Стоимость аренды'].value_counts().sort_values()

1495000.0       1
134999.0        1
22000.0 $       1
8081.0 €        1
55999.0         1
             ... 
50000.0      1235
35000.0      1305
nan          1317
45000.0      1441
40000.0      1483
Name: Стоимость аренды, Length: 601, dtype: int64

### 9. Санузел

In [42]:
#df['Санузел'].value_counts()

In [43]:
t2before = df.shape[0]

Удалим неизвестные значения '2.0'

In [44]:
df = df[df["Санузел"].str.contains('2.0')==False]
#df['Санузел'].value_counts()
t2after = df.shape[0]
t2 = t2before - t2after
print(f'Потеряли {t2} объявлений')

Потеряли 3744 объявлений


Разделим столбец:

In [45]:
df[['a', 'b']] = df['Санузел'].str.split(',', expand=True)

Заменим значения 'Совмещенный' на 0 и 'Раздельный' на 1:

In [46]:
df['a'] = df['a'].str.split('(')
df['a'] = df['a'].apply(lambda x: x[0])
df['a'] = df['a'].replace(['Совмещенный ', 'Раздельный '], [0, 1])

In [47]:
df.loc[df['b'].notnull(), 'b'] = 1
df.loc[df['b'].isnull(), 'b'] = 0

Сложим оба столбца (во втором только раздельный):

In [48]:
df['c'] = df['a'].astype(float) + df['b'].astype(float)
df['c'] = df['c'].astype(int)
df.drop(columns=['a', 'b'], inplace=True)

In [49]:
df = df.rename(columns={'c':'Раздельный туалет'})

In [50]:
df['Можно с детьми/животными'].value_counts()

Можно с детьми                       8570
Можно с детьми, Можно с животными    5942
Можно с животными                     204
Name: Можно с детьми/животными, dtype: int64

### 10. Можно с детьми/животными

Применим метод get_dummies:

In [51]:
df = pd.get_dummies(df, columns=['Можно с детьми/животными'])

In [52]:
df = df.rename(columns={'Можно с детьми/животными_Можно с детьми':'Можно с д',
                       'Можно с детьми/животными_Можно с детьми, Можно с животными':'Можно с д+ж',
                       'Можно с детьми/животными_Можно с животными':'Можно с ж'})

### 11. Окна

Применим метод get_dummies:

In [53]:
df = pd.get_dummies(df, columns=['Окна'])

In [54]:
df = df.rename(columns={'Окна_Во двор':'Окна во д',
                       'Окна_На улицу':'Окна на у',
                       'Окна_На улицу и двор':'Окна д+у'})

### 12. Ремонт

In [55]:
df['Ремонт'].value_counts()

Евроремонт       7223
Косметический    7021
Дизайнерский     2759
Без ремонта       121
Name: Ремонт, dtype: int64

In [56]:
#df['Ремонт'].isna().sum()

Удалим строки без информации о ремонте:

In [57]:
rbefore = df.shape[0]
df = df.dropna(subset = ['Ремонт'], axis=0)
rafter = df.shape[0]
r = rbefore - rafter
print(f'Потеряли {r} объявлений')

Потеряли 2511 объявлений


Заменим значения 'Без ремонта' на 0. 

In [58]:
df.loc[df['Ремонт'] == 'Без ремонта', 'Ремонт'] = 0

Применим метод get_dummies:

In [59]:
df = pd.get_dummies(df, columns=['Ремонт'])

In [60]:
df = df.rename(columns={'Ремонт_Дизайнерский':'Ремонт д',
                       'Ремонт_Евроремонт':'Ремонт е',
                       'Ремонт_Косметический':'Ремонт к'})
df = df.drop(columns='Ремонт_0')

### 13.  Балкон

In [61]:
df['Балкон'].value_counts()

Балкон (1)                5476
Лоджия (1)                4855
Балкон (1), Лоджия (1)     556
Лоджия (2)                 433
Балкон (2)                 329
Балкон (3)                  39
Лоджия (3)                  37
Балкон (1), Лоджия (2)      20
Балкон (2), Лоджия (2)      19
Балкон (2), Лоджия (1)      14
Балкон (3), Лоджия (1)       4
Балкон (1), Лоджия (3)       4
Лоджия (4)                   3
Балкон (4)                   3
Балкон (2), Лоджия (3)       2
Балкон (3), Лоджия (3)       1
Балкон (4), Лоджия (4)       1
Балкон (1), Лоджия (4)       1
Name: Балкон, dtype: int64

In [62]:
df['Балкон'].isna().sum()

5327

Предположим что отсутствующие значения означают отсутствие балкона. Заменим эти значения на ноль.

In [63]:
df['Балкон'] = df['Балкон'].fillna(0)

Разделим столбец:

In [64]:
df['Балкон'].value_counts()

Балкон (1)                5476
0                         5327
Лоджия (1)                4855
Балкон (1), Лоджия (1)     556
Лоджия (2)                 433
Балкон (2)                 329
Балкон (3)                  39
Лоджия (3)                  37
Балкон (1), Лоджия (2)      20
Балкон (2), Лоджия (2)      19
Балкон (2), Лоджия (1)      14
Балкон (1), Лоджия (3)       4
Балкон (3), Лоджия (1)       4
Лоджия (4)                   3
Балкон (4)                   3
Балкон (2), Лоджия (3)       2
Балкон (3), Лоджия (3)       1
Балкон (4), Лоджия (4)       1
Балкон (1), Лоджия (4)       1
Name: Балкон, dtype: int64

In [65]:
df[['a', 'b']] = df['Балкон'].str.split(', ', expand=True)

In [66]:
df['a'].value_counts()

Балкон (1)    6057
Лоджия (1)    4855
Лоджия (2)     433
Балкон (2)     364
Балкон (3)      44
Лоджия (3)      37
Балкон (4)       4
Лоджия (4)       3
Name: a, dtype: int64

In [67]:
df['a'].isna().sum()

5327

In [68]:
df['a'] = df['a'].str.split('(')
#df['a'] = df['a'].apply(lambda x: x[0])
df['a']


1                  NaN
2                  NaN
3                  NaN
4                  NaN
6                  NaN
             ...      
23363    [Балкон , 1)]
23364    [Лоджия , 1)]
23365    [Балкон , 1)]
23366    [Лоджия , 1)]
23367    [Лоджия , 1)]
Name: a, Length: 17124, dtype: object

В самом конце 

- переведем названия колонок на английском языке в одно или несколько слов с нижним подчёркиванием

- проверим что все значения внутри данных должны быть только численного типа (int или float)

In [69]:
df

Количество комнат       Метро  \
1                     4  Смоленская   
2                     4  Смоленская   
3                     4  Смоленская   
4                     2   Арбатская   
6                     5   Арбатская   
...                 ...         ...   
23363                 0    Говорово   
23364                 1    Солнцево   
23365                 2    Солнцево   
23366                 2    Солнцево   
23367                 2    Говорово   

                                                   Адрес Парковка      Балкон  \
1                          Москва, улица Новый Арбат, 27        1           0   
2                          Москва, улица Новый Арбат, 27        1           0   
3                       Москва, переулок Плотников, 21С1        1           0   
4                          Москва, улица Новый Арбат, 15        0           0   
6                          Москва, переулок Романов, 3С1        1           0   
...                                                  ...      ...         ...   
23363  Москва, Боровское шоссе, 2к7, ш. Боровское (3 ...        0  Балкон (1)   
23364  Москва, Производственная улица, 8к1, ш. Боровс...        0  Лоджия (1)   
23365                       Москва, Боровский проезд, 11        0  Балкон (1)   
23366  Москва, улица Богданова, 6к1, ш. Боровское (3 ...        1  Лоджия (1)   
23367  Москва, улица Богданова, 2к1, ш. Боровское (3 ...        0  Лоджия (1)   

                               Санузел  Мусоропровод  \
1      Совмещенный (2), Раздельный (1)           0.0   
2                      Совмещенный (3)           0.0   
3                      Совмещенный (3)           0.0   
4                      Совмещенный (2)           1.0   
6                      Совмещенный (2)           0.0   
...                                ...           ...   
23363                  Совмещенный (1)           0.0   
23364                  Совмещенный (1)           0.0   
23365                  Совмещенный (1)           0.0   
23366  Совмещенный (1), Раздельный (1)           1.0   
23367                   Раздельный (1)           1.0   

       Расстояние до метро, мин пешком  Общая площадь   Этаж  \
1                                  8.0          198.0   5/16   
2                                  7.0          200.0   5/16   
3                                  3.0          170.0    5/6   
4                                  7.0           58.0  12/26   
6                                  4.0          213.0    4/5   
...                                ...            ...    ...   
23363                              8.0           35.0  10/14   
23364                              7.0           38.7   5/18   
23365                              6.0           43.1    5/5   
23366                              1.0           52.5   8/23   
23367                              1.0           90.0   6/23   

                   Тип дома Стоимость аренды  Раздельный туалет  Можно с д  \
1       Монолитно-кирпичный         500000.0                  1          1   
2                      None         500000.0                  0          1   
3                      None         400000.0                  0          0   
4                 Панельный         225000.0                  0          0   
6                      None         350000.0                  0          0   
...                     ...              ...                ...        ...   
23363                  None          42000.0                  0          0   
23364            Монолитный          45000.0                  0          0   
23365             Кирпичный          50000.0                  0          1   
23366            Монолитный          55000.0                  1          0   
23367             Панельный          57000.0                  1          0   

       Можно с д+ж  Можно с ж  Окна во д  Окна на у  Окна д+у  Ремонт д  \
1                0          0          0          0         1         1   
2                0          0      